spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/onnx/Shape.scala

```diff
@@ -30,17 +31,23 @@ import scala.reflect.ClassTag
  * @tparam T The numeric type in this module parameters
  */
 class Shape[T: ClassTag](implicit ev: TensorNumeric[T])
-  extends Operation[Tensor[T], Tensor[T], T] {
+  extends TensorModule[T] {
 
   override def updateOutput(input: Tensor[T]): Tensor[T] = {
     val dimSize = input.nDimension()
+    Tensor[T]()
     output = Tensor[T](dimSize)
     (1 to dimSize).foreach(i => {
       output.setValue(i, ev.fromType(input.size(i)))
     })
     output
   }
 
+  // Shape if a constant layer,
+  // which means Output does not depend on Weight which gets updated by Gradient
+  override def updateGradInput(input: Tensor[T], gradOutput: Tensor[T]): Tensor[T] = {
+    gradInput
```

Comment:
I do not see why we need gradient in shape layer, since the output depends entirely on input.

Reply:
Because we need chain based backwarding, you need to pass `gradientInput` properly into proceeding layer. it's true the output depends entirely on input, so it would be zero, but expected to have the correct size.

Reply:
make its value zero explicitly,